### ISSUE TITLE
JDK-Suffix breaks semver contract

### ISSUE THREAD

**Issue by mrohlof-protofy** [ch4mpy/spring-addons#39]
**Describe the bug**
Not necessarily a bug in the library itself, but an inconvenience in the interaction with other tools:

I use this library as jdk-11 Dependency. Update mechanisms like dependabot think the jdk-17 variant is newer than jdk-11 and try to bump versions to it. Please think about re-structuring your naming scheme such that the jdk ist part of group or artifact id instead of version. I would prefer to see this:

```
    <groupId>com.c4-soft.springaddons</groupId>
    <artifactId>spring-security-oauth2-test-webmvc-addons-jdk11</artifactId>
    <version>3.1.16</version>
```

// Edit: This also goes for mvnrepository itself. See https://mvnrepository.com/artifact/com.c4-soft.springaddons/spring-security-oauth2-test-webmvc-addons/3.1.16-jdk11 for example

> Note: There is a new version for this artifact. New Version	3.1.16-jdk17

**Comment by ch4mpy** [comment]
I won't put the jdk version in the artifact name.

I'll give an other try to releasing with classifier.

anyhow, multi JDK build (which is a pain for me to do) will disapear when moving to spring-boot 3: builds will be done with JDK 17 only.

**Comment by ch4mpy** [comment]
Fixed with 3.2.0

`-jdk1.8` and `-jdk11` artifacts might drift in functionality JDK 17 flavor (one branch per JDK version => need to manually patch branches when updating master)
